source/simulator/assembly.cc

```diff
@@ -1624,14 +1644,37 @@ namespace aspect
     // TODO: Compute artificial viscosity once per timestep instead of each time
     // temperature system is assembled (as this might happen more than once per
     // timestep for iterative solvers)
-    const double nu
-      = compute_viscosity (scratch,
+
+    double nu = 0;
+	int count_additions = 0;
+	double nu_additions = 0.0;
+    if (use_smoothing == false)
+    {
+      nu = compute_viscosity (scratch,
```

Comment:
Yes, do it outside. For the moment, just write a loop over all cells where you compute the artificial viscosity for each element of a cell. I can show you later how to do the computation with WorkStream.